alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/jfx/events/mouse/DynamicMouseEventDispatcher.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.jfx.events.mouse
+
+import javafx.scene.input.MouseEvent
+
+/**
+ * A mouse event dispatcher which can receive temporary actions to listen to which will only be triggered once.
+ * These temporary actions have a higher priority than actions set through [action].
+ */
+open class DynamicMouseEventDispatcher : SimpleMouseEventDispatcher() {
+    private var dynamicActions: Map<MouseTriggerAction, (MouseEvent) -> Unit> = emptyMap()
+
+    override val listener = object : MouseActionListener {
+        override fun action(action: MouseTriggerAction, event: MouseEvent) {
+            if (action in dynamicActions.keys) {
+                dynamicActions[action]?.apply {
+                    invoke(event)
+                    dynamicActions = dynamicActions - action
+                }
+            } else {
+                triggers[action]?.invoke(event)
+            }
+        }
+    }
+
+    /**
+     * Set a dynamic action.
+     * @param trigger the trigger for the action
+     * @param job the job that will be run when the action occurs
+     */
+    fun setDynamicAction(trigger: MouseTriggerAction, job: (MouseEvent) -> Unit) {
+        dynamicActions = dynamicActions + (trigger to job)
```

Comment:
doesn't `+=` work?

Reply:
It does, but I get a very annoying warning that says "+= create new map under the hood" (spelling error included). So I thought it was preferable to avoid using `+=`.

Reply:
Well this does it as well, if `dynamicActions` is an immutable `Map` :)
Who is throwing the warning? Detekt? Does it make the build fail?